src/main/java/javaslang/collection/Iterator.java

```diff
@@ -128,6 +128,33 @@ public T next() {
     }
 
     /**
+     * Creates an Iterator which traverses along all given iterators.
+     * @param iterators The list of iterators
+     * @param <T>       Component type.
+     * @return A new {@code javaslang.collection.Iterator}
+     */
+    @SafeVarargs
+    static <T> Iterator<T> ofIterators(Iterator<T>... iterators) {
+        Objects.requireNonNull(iterators, "iterators is null");
+        List<Iterator<T>> queue = List.empty();
+        for (Iterator<T> iterator : iterators) {
+            queue = queue.prepend(iterator);
+        }
+        return queue.isEmpty() ? Iterator.empty() : new ConcatIterator<>(queue.reverse());
```

Comment:
Unfortunately this produces compilation warning 
```
Varargs method could cause heap pollution from non-reifiable varargs parameter iterators
```
and build fails. I'll change loop direction like in <code>List.of()</code>